cognite/neat/graph/extractors/_graph_capturing_sheet.py

```diff
@@ -166,35 +182,55 @@ def sheet2triples(
                 # Setting RDF type of the instance
                 if property_name == "identifier":
-                    triples.append((instance_namespace[row.identifier], RDF.type, model_namespace[sheet_name]))
+                    triples.append((instance_namespace[row.identifier], RDF.type, class_uri))
                     continue
                 elif not value:
                     continue
 
+                property_uri = (
+                    URIRef(str(class_property_pairs[sheet_name][property_name].source))
+                    if use_source_ids and class_property_pairs[sheet_name][property_name].source
+                    else model_namespace[property_name]
+                )
+
                 property_ = class_property_pairs[sheet_name][property_name]
-                is_one_to_many = (property_.max_count or 1) > 1 and separator
+
+                is_one_to_many = separator and (
+                    (property_.max_count and property_.max_count > 1) or not property_.max_count
+                )
+
                 values = value.split(separator) if is_one_to_many else [value]
 
                 # Adding object properties
                 if property_.property_type == "ObjectProperty":
                     triples.extend(
                         (
                             instance_namespace[row.identifier],
-                            model_namespace[property_name],
+                            property_uri,
                             instance_namespace[v.strip()],
                         )
                         for v in values
                     )
                 # Adding data properties
                 elif property_.property_type == "DatatypeProperty":
-                    triples.extend(
-                        (
-                            instance_namespace[row.identifier],
-                            model_namespace[property_name],
-                            Literal(v.strip(), datatype=XSD[property_.expected_value_type.suffix]),
-                        )
-                        for v in values
-                    )
+                    for v in values:
+                        try:
+                            triples.append(
+                                (
+                                    instance_namespace[row.identifier],
+                                    property_uri,
+                                    Literal(v.strip(), datatype=XSD[property_.expected_value_type.suffix]),
+                                )
+                            )
+                        except AttributeError:
+                            triples.append(
+                                (
+                                    instance_namespace[row.identifier],
+                                    property_uri,
+                                    Literal(v, datatype=XSD[property_.expected_value_type.suffix]),
+                                )
+                            )
```

Comment:
Lets pull out the only thing that can raise the error to make it more readable
```suggestion
                        try:
                            literal_value = v.strip()
                        except AttributeError:
                            literal_value = v
                            
                        triples.append(
                            (
                                instance_namespace[row.identifier],
                                property_uri,
                                Literal(literal_value , datatype=XSD[property_.expected_value_type.suffix]),
                            )
                        )
```